src/api/schema/events/metric.rs

```diff
@@ -0,0 +1,78 @@
+use async_graphql::Object;
+use chrono::{DateTime, Utc};
+use vector_core::event::metric::MetricTags;
+
+use super::EventEncodingType;
+use crate::{
+    config::OutputId,
+    event::{self},
+};
+
+#[derive(Debug, Clone)]
+pub struct Metric {
+    output_id: OutputId,
+    event: event::Metric,
+}
+
+impl Metric {
+    pub const fn new(output_id: OutputId, event: event::Metric) -> Self {
+        Self { output_id, event }
+    }
+}
+
+#[Object]
+/// Metric event with fields for querying metric data
+impl Metric {
+    /// Id of the component associated with the metric event
+    async fn component_id(&self) -> &str {
+        self.output_id.component.id()
+    }
+
+    /// Metric timestamp
+    async fn timestamp(&self) -> Option<&DateTime<Utc>> {
+        self.event.data().timestamp().as_ref()
+    }
+
+    /// Metric name
+    async fn name(&self) -> &str {
+        self.event.name()
+    }
+
+    /// Metric namespace
+    async fn namespace(&self) -> Option<&str> {
+        self.event.namespace()
+    }
+
+    /// Metric kind
+    async fn kind(&self) -> &str {
+        match self.event.kind() {
+            event::MetricKind::Incremental => "incremental",
+            event::MetricKind::Absolute => "absolute",
+        }
+    }
```

Comment:
We may want to make this a GraphQL enum type rather than string.

Reply:
Changed in [d4eed34](https://github.com/vectordotdev/vector/pull/11201/commits/d4eed349868db5cba309a8dc49c7e874d95a121d)